src/update_engine/delta_diff_generator.cc

```diff
@@ -1243,6 +1353,8 @@ bool DeltaDiffGenerator::GenerateDeltaUpdateFile(
 
   LOG(INFO) << "Reading files...";
 
+  vector<InstallOperation> kernel_ops;
```

Comment:
Oh right, generator.Add is FullUpdateGenerator::Add, and it takes the place of DeltaCompressKernel when generating full updates. So it is free to output multiple operations if it wants, specifically it creates one operation per 10MB chunk.